File: backend/src/DailyActivityTracker.Domain/Entities/Activity.cs
using DailyActivityTracker.Domain.Common;
using DailyActivityTracker.Domain.Enums;

namespace DailyActivityTracker.Domain.Entities;

public class Activity : BaseEntity
{
    public Guid UserId { get; set; }

    public User User { get; set; } = null!;

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public DateTime ActivityDate { get; set; }

    public ActivityStatus Status { get; set; }

    public ActivityPriority Priority { get; set; }
}

File: frontend/src/types/activity.ts
export type ActivityStatus = 
    | 1 //'Pending'
    | 2 //'InProgress'
    | 3 //'Completed'
    | 4 //'Cancelled'

export type ActivityPriority =
    | 1 //'Low'
    | 2 //'Medium'
    | 3 //'High'

export type Activity = {
    id: string
    userId: string
    title: string
    description: string | null
    activityDate: string
    status: ActivityStatus
    priority: ActivityPriority
}